tests/core/test_device.py

```diff
@@ -117,3 +121,66 @@ async def test_device_log_has_correct_name():
     assert device.log.extra["ophyd_async_device_name"] == ""
     device.set_name("device")
     assert device.log.extra["ophyd_async_device_name"] == "device"
+
+
+async def test_device_lazily_connects(RE):
+    async with DeviceCollector(sim=True, connect=False):
+        sim_motor = motor.Motor("BLxxI-MO-TABLE-01:X")
+
+    assert sim_motor._connect_task is None
+
+    # When ready to connect
+    RE(ensure_connected(sim_motor, sim=True))
+
+    assert (
+        sim_motor._connect_task
+        and sim_motor._connect_task.done()
+        and not sim_motor._connect_task.exception()
+    )
+
+
+class MotorBundle(Device):
+    def __init__(self, name: str) -> None:
+        self.X = motor.Motor("BLxxI-MO-TABLE-01:X")
+        self.Y = motor.Motor("BLxxI-MO-TABLE-01:Y")
+        self.V: DeviceVector[motor.Motor] = DeviceVector(
+            {
+                0: motor.Motor("BLxxI-MO-TABLE-21:X"),
+                1: motor.Motor("BLxxI-MO-TABLE-21:Y"),
+                2: motor.Motor("BLxxI-MO-TABLE-21:Z"),
+            }
+        )
+
+
+async def test_device_with_children_lazily_connects(RE):
+    parentMotor = MotorBundle("parentMotor")
+
+    assert (
+        parentMotor._connect_task is None
+        and parentMotor.X._connect_task is None
+        and parentMotor.Y._connect_task is None
+    )
+
+    RE(ensure_connected(parentMotor, sim=True))
+
+    assert (
+        parentMotor.X._connect_task
+        and parentMotor.X._connect_task.done()
+        and not parentMotor.X._connect_task.exception()
+    )
+    assert (
+        parentMotor.Y._connect_task
+        and parentMotor.Y._connect_task.done()
+        and not parentMotor.Y._connect_task.exception()
+    )
+    assert (
+        parentMotor._connect_task
+        and parentMotor._connect_task.done()
+        and not parentMotor._connect_task.exception()
+    )
+    for vector in parentMotor.V.values():
+        assert (
+            vector._connect_task
+            and vector._connect_task.done()
+            and not vector._connect_task.exception()
+        )
```

Comment:
In general I try and avoid copy paste, even in tests...
```suggestion
    for device in [parentMotor, parentMotor.X, parentMotor.Y] + list(parentMotor.V.values()):
        assert (
            device._connect_task
            and device._connect_task.done()
            and not device._connect_task.exception()
        )
```